projects/packages/connection/src/class-manager.php

```diff
@@ -2593,4 +2603,263 @@ public static function is_ready_for_cleanup( $current_plugin_slug ) {
 		return empty( $active_plugins ) || ! is_array( $active_plugins )
 			|| ( count( $active_plugins ) === 1 && array_key_exists( $current_plugin_slug, $active_plugins ) );
 	}
+
+	/**
+	 * Verifies the request by checking the signature
+	 *
+	 * @since jetpack-4.6.0 Method was updated to use `$_REQUEST` instead of `$_GET` and `$_POST`. Method also updated to allow
+	 * passing in an `$environment` argument that overrides `$_REQUEST`. This was useful for integrating with SSO.
+	 * @since $$next-version$$ Ported from Jetpack to the Connection package.
+	 *
+	 * @param null|array $environment Value to override $_REQUEST.
+	 *
+	 * @return void
+	 */
+	public function verify_json_api_authorization_request( $environment = null ) {
```

Comment:
That makes sense. Done in 20e4f2ede9477572b0860b5daf5523810ea17344

I opted for a new class, since it's used outside of SSO, like when used by third-parties using the JSON API module to authenticate with the site.